### ISSUE TITLE
`customize {  windows_options { domain_ou` path containing spaces fails with error

### ISSUE THREAD

**Issue by mrdevops100** [hashicorp/terraform-provider-vsphere#2206]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.7.5

### Terraform Provider

v2.8.1

### VMware vSphere

v7.0.1

### Description

I am creating vm using the vsphere template and trying to use the new feature "domain_ou" to assign the computer to an OU but post provisioning the new server does not land under correct OU. It automatically falls under the general "Computers" OU in the active directory.

### Affected Resources or Data Sources

resource/vsphere_virtual_machine

### Terraform Configuration

n/a

### Debug Output

n/a

### Expected Behavior

Computer should be assigned the right OU

### Actual Behavior

It falls under the global "Computers" OU

### Steps to Reproduce

Create new VM using vsphere_virtual_machine and use the domain_ou option

### References

n/a

**Comment by tenthirtyam** [comment]
cc @ultral who wrote the PR.

**Comment by ultral** [comment]
@mrdevops100 could you please share some info?
1. snippet for VM creation
2. terrafrom debug output
3. your OUs structure

**Comment by mrdevops100** [comment]
@ultral 

OU Structure: "OU=Domain Servers,DC=mycompany,DC=com"
<img width="271" alt="vmsnip" src="https://github.com/hashicorp/terraform-provider-vsphere/assets/144401428/9b5e4b8f-c9bd-40ad-a407-5b9711d15782">


Please find the attached debug log.
[debug_output.txt](https://github.com/hashicorp/terraform-provider-vsphere/files/15484615/debug_output.txt)

**Comment by ultral** [comment]
@mrdevops100  could you please attach the terraform code also and vmware version?

**Comment by tenthirtyam** [comment]
@mrdevops100 - pleas share a redacted form of your configuration for review and reproduction purposes.

**Comment by D1StrX** [comment]
The issue is that it doesn't properly handle the space in _OU=ou name_

**Comment by ultral** [comment]
@D1StrX could you please share

1. your terraform snippet for VM creation
2. terrafrom debug output
3. your OUs structure
4. vmware version
5. guest OS version

**Comment by D1StrX** [comment]
Tested with the same setup and outcome/output as https://github.com/hashicorp/terraform-provider-vsphere/issues/2209.
Guest OS version in test was windows2019srv_64Guest

**Comment by tenthirtyam** [comment]
Could you try:

```heck
domain_ou = "'OU=Workstations,OU=SDE,OU=Business Units,DC=sensitive,DC=com'"
```

**Comment by D1StrX** [comment]
That unfortunately doesn't work. Same error:
`Error: error sending customization spec: A specified parameter was not correct: The Domain OU string must refer to an existing OU in your target Domain. Please don't specify any other values excepting OU and DC in it.`

**Comment by ultral** [comment]
rough suggestion is that it might space related issue. in both cases OU contains spaces: `OU=Business Units` & `OU=Domain Servers`. will try to reproduce that

**Comment by tenthirtyam** [comment]
I have reproduced this issue with the following:

🔴 **FAIL**: `domain_ou = "OU=Business Units,DC=rainpole,DC=io"`
🟢 **PASS**: `domain_ou = "OU=BusinessUnits,DC=rainpole,DC=io"`

One issue O noticed was in `func flattenWindowsOptions(customizationPrep` where `domain_OU` was used instead of `domain_ou`. Go is case sensitive; however, that **does not** resolve the issue.

https://github.com/hashicorp/terraform-provider-vsphere/blob/0f5383d22b6d3b0c9af47cb807bcdb14b1a20c70/vsphere/internal/helper/guestoscustomizations/customizations_helper.go#L443-L445

**Comment by tenthirtyam** [comment]
Also. of note, per the OP @mrdevops100 is using vSphere 7.x, but this capability is only available in vSphere 8.0.2 (Update 2) and later.

**Comment by tenthirtyam** [comment]
I can confirm that this **is not** an issue with the provider, but rather an issue in vSphere.

When setting the OU Path in a customiation specification for Windows, the results are the same.

I have reproduced this issue with the following:

🔴 **FAIL**: "_OU path: OU=Business Units,DC=rainpole,DC=io_"

![image](https://github.com/hashicorp/terraform-provider-vsphere/assets/7771363/11e9a3d7-44af-47e3-bef8-03340ca66315)

🟢 **PASS**: "_OU path: OU=BusinessUnits,DC=rainpole,DC=io_"

![image](https://github.com/hashicorp/terraform-provider-vsphere/assets/7771363/1519131b-f1a4-4ae4-8818-8ba22304bf25)

I'll do the following:

- Open an internal bug for vSphere customization specifications.
- Update the documentation for 2.8.2 to note that:
  - OUs with spaces are not accepted.
  - vSphere 8.0.2 or later is required for this option.

Once this is completed this bug will be closed due to the upstream issue in vSphere.

**Ryan Johnson**
Distinguished Engineer, VMware by Broadcom
